rotkehlchen/api/rest.py

```diff
@@ -2320,6 +2322,84 @@ def remove_queried_address_per_module(
 
+    def get_lido_csm_node_operators(self) -> Response:
+        return api_response(
+            _wrap_in_ok_result([
+                {
+                    'address': entry.address,
+                    'node_operator_id': entry.node_operator_id,
+                    'metrics': entry.metrics.serialize() if entry.metrics else None,
+                }
+                for entry in DBLidoCsm(self.rotkehlchen.data.db).get_node_operators()
+            ]),
+            status_code=HTTPStatus.OK,
+        )
+
+    def add_lido_csm_node_operator(
+            self,
+            address: ChecksumEvmAddress,
+            node_operator_id: int,
+    ) -> Response:
+        try:
+            DBLidoCsm(self.rotkehlchen.data.db).add_node_operator(
+                address=address,
+                node_operator_id=node_operator_id,
+            )
+        except InputError as e:
+            return api_response(wrap_in_fail_result(str(e)), status_code=HTTPStatus.CONFLICT)
+
+        # Compute and persist metrics for the newly added operator. If it fails
+        # we still return the list but metrics will be empty until refreshed.
+        try:
+            metrics = LidoCsmMetricsFetcher(
+                evm_inquirer=self.rotkehlchen.chains_aggregator.ethereum.node_inquirer,
+            ).get_operator_stats(node_operator_id)
+            DBLidoCsm(self.rotkehlchen.data.db).set_metrics(node_operator_id, metrics)
+        except RemoteError as e:
+            log.error(
+                f'Failed to fetch Lido CSM metrics for new operator {node_operator_id}: {e}',
+            )
+
+        return self.get_lido_csm_node_operators()
+
+    def remove_lido_csm_node_operator(
+            self,
+            address: ChecksumEvmAddress,
+            node_operator_id: int,
+    ) -> Response:
+        try:
+            DBLidoCsm(self.rotkehlchen.data.db).remove_node_operator(
+                address=address,
+                node_operator_id=node_operator_id,
+            )
+        except InputError as e:
+            return api_response(wrap_in_fail_result(str(e)), status_code=HTTPStatus.CONFLICT)
+
+        return self.get_lido_csm_node_operators()
+
+    def refresh_lido_csm_metrics(self) -> Response:
+        """Recompute metrics for a given node operator id,
+        or for all tracked operators if omitted."""
+        metrics_fetcher = LidoCsmMetricsFetcher(
+            evm_inquirer=self.rotkehlchen.chains_aggregator.ethereum.node_inquirer,
+        )
+
+        result = []
+        for entry in DBLidoCsm(self.rotkehlchen.data.db).get_node_operators():
+            try:
+                metrics = metrics_fetcher.get_operator_stats(entry.node_operator_id)
+                metrics_payload = metrics.serialize()
+                DBLidoCsm(self.rotkehlchen.data.db).set_metrics(entry.node_operator_id, metrics)
+            except RemoteError as e:
+                log.error(f'Failed to refresh Lido CSM metrics for {entry}: {e}')
```

Comment:
As also mentioned above perhaps use a specific status code return and handle in frontend to give a specific user warning at api return?

The reason for this is that logs are almost always ignored unless we know to check logs.

1. The user needs to set logs to error/debug. They are off by default (due to wanting to avoid any data leaking in plaintext at the user's machine).
2. Most users would have no idea an error occurred. Would just tell us something does not work. And then we would do the: "Please open debug logs", "Send us the logs", "Look at them", "find the error" etc. While all this could be avoided by a proper error handling here.